_posts/en/newsletters/2021-09-15-newsletter.md

```diff
@@ -0,0 +1,209 @@
+---
+title: 'Bitcoin Optech Newsletter #166'
+permalink: /en/newsletters/2021/09/15/
+name: 2021-09-15-newsletter
+slug: 2021-09-15-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a new proposal for a covenant opcode
+and summarizes a request for feedback on implementing regular reorgs on
+signet.  Also included are our regular sections with ideas for preparing
+for taproot activation, a list of new releases and release candidates,
+and descriptions of popular Bitcoin infrastructure software.
+
+## News
+
+- **Covenant opcode proposal:** Anthony Towns posted to the Bitcoin-Dev
+  mailing list the [overview][towns overview] of an idea for a covenant
+  opcode, plus a more [technically detailed][towns detailed] message
+  describing how the opcode (and some other tapscript changes) would
+  work.
+
+    In short, a new `OP_TAPLEAF_UPDATE_VERIFY` opcode (TLUV) takes
+    information about the taproot input being spent, makes modifications
+    described in a tapscript, and requires that the result be equivalent
+    to the scriptPubKey in the output in the same position as the input.
+    This allows TLUV to constrain where the bitcoins can be spent (the
+    definition of a [covenant][topic covenants]), similar to other
+    proposed opcodes such as [OP_CHECKSIGFROMSTACK][topic
+    op_checksigfromstack] (CSFS) and [OP_CHECKTEMPLATEVERIFY][topic
+    op_checktemplateverify] (CTV).  Where it differs from previous
+    proposals are the modifications it allows the creator of the
+    tapscript to make:
+
+    - **Internal key tweak:** every taproot address commits to an
+      internal key that (if its individual or aggregated private key is
```

Comment:
"if its ... aggregated private key is known" seems inaccurate to me. In key aggregation schemes like MuSig, the constituent private keys are used to create partial signatures, which are aggregated to create a complete signature. The private keys themselves are not aggregated.